core/domain/exp_domain.py

```diff
@@ -2964,6 +2967,65 @@ def delete_state(self, state_name):
 
         del self.states[state_name]
 
+    def get_trainable_states_dict(self, old_states, change_list):
+        """Retrieves the state names of all trainable states in an exploration
+        segregated into state names with changed and unchanged answer groups.
+
+        Args:
+            old_states: dict. Dictionary containing all State domain objects.
+            change_list: list(dict). A list of changes introduced in this
+                commit.
+
+        Returns:
+            dict. The trainable states dict. This dict has two keys representing
+                state names with changed and unchanged answer groups
+                respectively.
+        """
+        trainable_states_dict = {
+            'state_names_with_changed_answer_groups': [],
+            'state_names_with_unchanged_answer_groups': []
+        }
+        new_states = self.states
+        for state_name in new_states:
+            new_state = new_states[state_name]
+            if not new_state.can_undergo_classification():
+                continue
+
+            old_state_name = state_name
+            renamed_cmds = []
```

Comment:
We should not be duplicating code. Perhaps pull this part out into a function in exp_domain that is referenced in both places, or do the computation here and return the mapping somehow so that the computation need not be done again. This may involve changing the name and description of the method accordingly.